test/server/cards/12-PV/CrypticCollapse.spec.js

```diff
@@ -37,5 +40,97 @@ describe('Cryptic Collapse', function () {
             expect(this.player2.amber).toBe(4);
             expect(this.player1).toHavePrompt('Choose a card to play, discard or use');
         });
+
+        /**
+         * Regression test for https://github.com/keyteki/keyteki/issues/4663
+         */
+        it('should not re-discard cards already discarded', function () {
+            // Set things up so that the 2 Hazard Zerps are in hand, with
+            // nothing else but Cryptic Collapse.
+            this.player1.scrap(this.emberImp);
+            this.player1.scrap(this.parasiticArachnoid);
+            this.player1.moveCard(this.hazardZerp1, 'hand');
+            this.player1.moveCard(this.hazardZerp2, 'hand');
+
+            this.player1.play(this.crypticCollapse);
+
+            // First we lay out the discard order for the Cryptic Collapse
+            expect(this.player1).toHavePrompt('Cryptic Collapse');
+            expect(this.player1).toHavePrompt('Select next card to discard');
+            expect(this.player1).toBeAbleToSelect(this.hazardZerp1);
+            expect(this.player1).toBeAbleToSelect(this.hazardZerp2);
+            this.player1.clickCard(this.hazardZerp1);
+
+            expect(this.player1).toHavePrompt('Cryptic Collapse');
+            expect(this.player1).toHavePrompt('Select next card to discard');
```

Comment:
Hmm this doesn't seem right to me.  You should have to completely resolve hazardZerp1's effect before choosing the next card to discard, I think (by the rulebook).  What if the scrap effect caused you to draw a card (Brillix Ponder)?

(I am open to this being a partial measure that gets us closer to the right solution for some common cases. Just want to make sure I understand if this is intentional or not.)

Reply:
Ok thanks, fixing this in a followup makes sense to me.  Thank you for working on this!

> if I discard Brillix Ponder as my last card when doing Cryptic Collapse, does the drawn card get discarded as well?

I don't think we have an official ruling here.  What I've seen in the discord rules discussions (e.g. [here](https://discord.com/channels/563737400275107920/588586239997902858/1411837430222295080)) is that "discard your hand" is usually interpreted as discarding one at a time until you have 0 cards in hand (so the drawn card would get discarded).